coorx/base_transform.py

```diff
@@ -356,8 +366,28 @@ def __getstate__(self):
     def __setstate__(self, state):
         from_cs, to_cs, graph = state.pop('_systems', (None, None, None))
         self.__dict__.update(state)
-        self._systems = (None, None)
-        self.set_systems(from_cs, to_cs, graph)
+        if not self.Dependent:
+            self._systems = (None, None)
+            self.set_systems(from_cs, to_cs, graph)
+
+    def copy(self, from_cs=None, to_cs=None):
+        """Return a copy of this transform.
+        """
+        tr = self.__class__(dims=self.dims)
+        state = self.__getstate__()
+        if from_cs is not None or to_cs is not None:
+            if self.Dependent:
```

Comment:
I think a better way to achieve thi might be to have individual subclasses like InverseTransform implement their own getstate/setstate in a way that handles whether coordinate systems are allowed here